cmd/kueuectl/app/list/list_clusterqueue.go

```diff
@@ -0,0 +1,252 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	kueuev1beta1 "sigs.k8s.io/kueue/client-go/clientset/versioned/typed/kueue/v1beta1"
+)
+
+var (
+	cqShort   = `List ClusterQueues`
+	cqLong    = templates.LongDesc(`Lists all ClusterQueues, potentially limiting output to those that are active/inactive and matching the label selector.`)
+	cqExample = templates.Examples(`
+		# List ClusterQueue
+		kueuectl list clusterqueue`)
+)
+
+type clusterQueueActive string
+
+const (
+	// No filter
+	clusterQueueActiveAll   clusterQueueActive = "*"
+	clusterQueueActiveEmpty clusterQueueActive = ""
+
+	// Truthy values
+	clusterQueueActiveTrue clusterQueueActive = "true"
+
+	// Falsy values
+	clusterQueueActiveFalse clusterQueueActive = "false"
+)
+
+type ClusterQueueOptions struct {
+	// PrintFlags holds options necessary for obtaining a printer
+	PrintFlags *genericclioptions.PrintFlags
+
+	LabelSelector string
+	FieldSelector string
+
+	// Kueuectl flags
+	// Active is the flag to filter */true/false (all/active/inactive) cluster queues
+	// Active means the cluster queue has kueue.ClusterQueueActive condition with status=metav1.ConditionTrue
+	Active string
+
+	Client kueuev1beta1.KueueV1beta1Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	// ctx is the context for the command
+	ctx context.Context
```

Comment:
let's avoid caching contexts when possible.  In this case we should not do it, so drop this field.